tools/analyzer_plugin/test/integration/matchers.dart

```diff
@@ -0,0 +1,149 @@
+import 'package:analyzer/src/generated/source.dart';
+import 'package:analyzer_plugin/protocol/protocol_common.dart';
+import 'package:analyzer_plugin/protocol/protocol_generated.dart';
+import 'package:analyzer_plugin/utilities/assist/assist.dart';
+import 'package:over_react_analyzer_plugin/src/diagnostic_contributor.dart';
+import 'package:test/test.dart';
+
+import 'test_bases/server_plugin_contributor_test_base.dart';
+
+/// Returns a [Matcher] that verifies that the item is a source change with a
+/// priority and ID matching [assistKind].
+///
+/// If [locatedAt] is non-null, the analysis error's location (offset and
+/// length) matches it
+TypeMatcher<PrioritizedSourceChange> isAssist(AssistKind assistKind, {SourceSelection locatedAt}) {
```

Comment:
Just realized it doesn't make sense for these `isAssist()` and `isFix()` matchers to support the `locatedAt` option because `PrioritizedSourceChange`s don't target a single selection range like an error does.. they can have multiple file edits and a resulting selection position that the cursor should be placed at. Going to remove `locatedAt` for these two and instead tests should just verify that applying the assist/fix results in the expected source contents.